### PR TITLE
Port edit fns

### PR DESCRIPTION
**Pull request by shanavas786**

### CODE REVIEW

File: rust_src/src/editfns.rs [hunk 1]
```diff
@@ -49,3 +50,49 @@ pub fn bobp() -> LispObject {
...
+    }
+
+    let m = marker_position(current_buf.mark());
+    if m.is_nil() {
+        error!("The mark is not set now, so there is no region");
+    }
+
+    let num = m.as_fixnum_or_error();
+    // Clip to the current narrowing (bug#11770)
+    LispObject::from_fixnum(if ((current_buf.pt as EmacsInt) < num) == beginningp {
```
Comment (DavidDeSimone) [inline rust_src/src/editfns.rs]: I would personally find this more readable if we pulled the if statement out of the from_fixnum call

File: src/lisp.h [hunk 2]
```diff
@@ -1007,11 +1007,6 @@ INLINE bool
 #define FIXNUM_OVERFLOW_P(i) \
   (! ((0 <= (i) || MOST_NEGATIVE_FIXNUM <= (i)) && (i) <= MOST_POSITIVE_FIXNUM))
 
-INLINE ptrdiff_t
-clip_to_bounds (ptrdiff_t lower, EMACS_INT num, ptrdiff_t upper)
-{
-  return num < lower ? lower : num <= upper ? num : upper;
-}
```
Comment (Wilfred) [inline src/lisp.h]: Please leave a function declaration here. This has added a new warning when compiling the C:

```c
  CC       xdisp.o

xdisp.c: In function ‘init_iterator’:

xdisp.c:2821:9: warning: implicit declaration of function ‘clip_to_bounds’ [-Wimplicit-function-declaration]

         (-1, XINT (BVAR (current_buffer, selective_display)),
```

Comment (shanavas786) [reply src/lisp.h]: thanks, updated

**Comment by shanavas786** [comment]
rebased

**Comment by shanavas786** [comment]
conflicts resolved

**Comment by shanavas786** [comment]
travis builds for mac was unexpectedly canceled. Ready to be merged now

**Comment by Wilfred** [comment]
Awesome! Looks like #298 has introduced merge conflicts (sorry about that), but we can merge once those are sorted 👍

**Pull request merged**